src/Microsoft.CodeAnalysis.BannedApiAnalyzers/UnitTests/RestrictedInternalsVisibleToAnalyzerTests.cs

```diff
@@ -0,0 +1,1864 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CSharp.BannedApiAnalyzers;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Testing;
+using Microsoft.CodeAnalysis.VisualBasic.BannedApiAnalyzers;
+using Test.Utilities;
+using Xunit;
+
+namespace Microsoft.CodeAnalysis.BannedApiAnalyzers.UnitTests
+{
+    public class RestrictedInternalsVisibleToAnalyzerTests : DiagnosticAnalyzerTestBase
```

Comment:
💡 All new analyzer tests should be using the new test framework

Reply:
I tried to initially but it didn’t support the required structure here - 2 input source files in different projects with a P2P reference and validate diagnostics in referencing project. I think it is fine to use custom helpers for such custom analyzers. In general, we should certainly try to use new test framework as much as possible.

Reply:
I'm working to remove all of the custom test infrastructure. So far the code fix testing support has all been removed, and the analyzer test support is all that's left. I agree the test support for this analyzer could be improved, but here's an example showing how it can be done:

https://github.com/dotnet/machinelearning/blob/master/test/Microsoft.ML.CodeAnalyzer.Tests/Code/BestFriendTest.cs